climada/util/api_client.py

```diff
@@ -254,19 +253,14 @@ class NoResult(Exception):
     class NoConnection(Exception):
         """To be raised if there is no internet connection and no cached result."""
 
+    @property
+    def online(self) -> bool:
+        """Check if this client can connect to the target URL"""
+        # Use just the base location
+        parse_result = urlsplit(self.url)
+        query_url = urlunsplit((parse_result.scheme, parse_result.netloc, "", "", ""))
 
-    @staticmethod
-    def _is_online(url):
-        host = [x for x in url.split('/')
-                if x not in ['https:', 'http:', '']][0]
-        port = 80 if url.startswith('http://') else 443
-        socket.setdefaulttimeout(1)
-        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as skt:
-            try:
-                skt.connect((host, port))
-                return True
-            except socket.error:
-                return False
+        return requests.head(query_url, timeout=1).status_code == 200
```

Comment:
Seems like the `timeout` argument doesn't seem to have an effect when simply checking for a connection.
- on Windows (and WSL) the latency period is about 10 seconds regardless of the `timeout` value
- on a Mac (?) the response is immediate also regardless of the `timeout` value - right?

I think we should remove the argument and instead add an inline comment about the unexpected timeout behavior of `requests` for future developers.

Reply:
```suggestion

        # NOTE: 'timeout' might not work as intended, depending on OS and network status
        return requests.head(query_url, timeout=1).status_code == 200
```